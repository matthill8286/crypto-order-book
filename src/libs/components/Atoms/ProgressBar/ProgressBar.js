function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import * as React from 'react';
import { styled } from "../../../styles/styled";
var StyledProgressBar = styled.progress.withConfig({
  displayName: "ProgressBar__StyledProgressBar",
  componentId: "iekkgs-0"
})(["display:inline-block;appearance:none;position:relative;height:", ";border:none;background:", ";background-size:auto;width:100%;overflow:hidden;color:", ";&::-webkit-progress-value{background:", ";}&::-webkit-progress-bar{background:", ";}&::-moz-progress-bar{background:", ";}&::-ms-fill{background:", ";}"], function (_ref) {
  var isSmall = _ref.isSmall;
  return isSmall ? '4px' : '15px';
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.color.grey2;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.color.primary;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.color.primary;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.color.grey2;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.color.primary;
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.color.primary;
});
export var ProgressBar = function ProgressBar(_ref8) {
  var value = _ref8.value,
      _ref8$max = _ref8.max,
      max = _ref8$max === void 0 ? 100 : _ref8$max,
      _ref8$small = _ref8.small,
      small = _ref8$small === void 0 ? false : _ref8$small,
      otherProps = _objectWithoutProperties(_ref8, ["value", "max", "small"]);

  return React.createElement(StyledProgressBar, _extends({
    max: max,
    value: value,
    isSmall: small
  }, otherProps));
};
//# sourceMappingURL=ProgressBar.js.map