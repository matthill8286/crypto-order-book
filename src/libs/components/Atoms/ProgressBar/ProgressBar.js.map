{"version":3,"sources":["../../../../src/components/Atoms/ProgressBar/ProgressBar.tsx"],"names":["React","styled","StyledProgressBar","progress","isSmall","theme","color","grey2","primary","ProgressBar","value","max","small","otherProps"],"mappings":";;;;;;AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AACA,SAASC,MAAT;AAYA,IAAMC,iBAAiB,GAAGD,MAAM,CAACE,QAAV;AAAA;AAAA;AAAA,0TAIX;AAAA,MAAGC,OAAH,QAAGA,OAAH;AAAA,SAAkBA,OAAO,GAAG,KAAH,GAAW,MAApC;AAAA,CAJW,EAMP;AAAA,MAAGC,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYC,KAA3B;AAAA,CANO,EAUZ;AAAA,MAAGF,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYE,OAA3B;AAAA,CAVY,EAaL;AAAA,MAAGH,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYE,OAA3B;AAAA,CAbK,EAiBL;AAAA,MAAGH,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYC,KAA3B;AAAA,CAjBK,EAqBL;AAAA,MAAGF,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYE,OAA3B;AAAA,CArBK,EAyBL;AAAA,MAAGH,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,KAAN,CAAYE,OAA3B;AAAA,CAzBK,CAAvB;AA6BA,OAAO,IAAMC,WAAuC,GAAG,SAA1CA,WAA0C;AAAA,MACrDC,KADqD,SACrDA,KADqD;AAAA,wBAErDC,GAFqD;AAAA,MAErDA,GAFqD,0BAE/C,GAF+C;AAAA,0BAGrDC,KAHqD;AAAA,MAGrDA,KAHqD,4BAG7C,KAH6C;AAAA,MAIlDC,UAJkD;;AAAA,SAKpC,oBAAC,iBAAD;AAAmB,IAAA,GAAG,EAAEF,GAAxB;AAA6B,IAAA,KAAK,EAAED,KAApC;AAA2C,IAAA,OAAO,EAAEE;AAApD,KAA+DC,UAA/D,EALoC;AAAA,CAAhD","sourcesContent":["import * as React from 'react'\nimport { styled } from '@/styles/styled'\n\nexport interface ProgressBarProps {\n  value: number\n  max?: number\n  small?: boolean\n}\n\nexport interface StyledProgressProps {\n  isSmall: boolean\n}\n\nconst StyledProgressBar = styled.progress<StyledProgressProps>`\n  display: inline-block;\n  appearance: none;\n  position: relative;\n  height: ${({ isSmall }) => (isSmall ? '4px' : '15px')};\n  border: none;\n  background: ${({ theme }) => theme.color.grey2};\n  background-size: auto;\n  width: 100%;\n  overflow: hidden;\n  color: ${({ theme }) => theme.color.primary};\n\n  &::-webkit-progress-value {\n    background: ${({ theme }) => theme.color.primary};\n  }\n\n  &::-webkit-progress-bar {\n    background: ${({ theme }) => theme.color.grey2};\n  }\n\n  &::-moz-progress-bar {\n    background: ${({ theme }) => theme.color.primary};\n  }\n\n  &::-ms-fill {\n    background: ${({ theme }) => theme.color.primary};\n  }\n`\n\nexport const ProgressBar: React.FC<ProgressBarProps> = ({\n  value,\n  max = 100,\n  small = false,\n  ...otherProps\n}): JSX.Element => <StyledProgressBar max={max} value={value} isSmall={small} {...otherProps} />\n"],"file":"ProgressBar.js"}