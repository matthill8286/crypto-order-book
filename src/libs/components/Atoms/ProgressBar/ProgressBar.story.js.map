{"version":3,"sources":["../../../../src/components/Atoms/ProgressBar/ProgressBar.story.tsx"],"names":["boolean","number","storiesOf","React","ProgressBar","module","add","value","small"],"mappings":"AAAA,SAASA,OAAT,EAAkBC,MAAlB,QAAgC,wBAAhC;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEAF,SAAS,CAAC,iCAAD,EAAoCG,MAApC,CAAT,CAAqDC,GAArD,CAAyD,SAAzD,EAAoE,YAAM;AACxE,MAAMC,KAAK,GAAGN,MAAM,CAAC,OAAD,EAAU,EAAV,CAApB;AACA,MAAMO,KAAK,GAAGR,OAAO,CAAC,eAAD,EAAkB,KAAlB,CAArB;AAEA,SAAO,oBAAC,WAAD;AAAa,IAAA,KAAK,EAAEO,KAApB;AAA2B,IAAA,KAAK,EAAEC;AAAlC,IAAP;AACD,CALD","sourcesContent":["import { boolean, number } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { ProgressBar } from './ProgressBar'\n\nstoriesOf('Design System/Atoms/ProgressBar', module).add('Default', () => {\n  const value = number('Value', 50)\n  const small = boolean('Small Version', false)\n\n  return <ProgressBar value={value} small={small} />\n})\n"],"file":"ProgressBar.story.js"}