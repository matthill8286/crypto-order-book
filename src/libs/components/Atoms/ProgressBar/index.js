export * from "./ProgressBar";
//# sourceMappingURL=index.js.map