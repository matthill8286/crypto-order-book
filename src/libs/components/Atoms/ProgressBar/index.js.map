{"version":3,"sources":["../../../../src/components/Atoms/ProgressBar/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './ProgressBar'\n"],"file":"index.js"}