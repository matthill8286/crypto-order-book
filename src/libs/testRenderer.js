import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map