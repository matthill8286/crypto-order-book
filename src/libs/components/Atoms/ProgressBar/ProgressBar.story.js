import { boolean, number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { ProgressBar } from "./ProgressBar";
storiesOf('Design System/Atoms/ProgressBar', module).add('Default', function () {
  var value = number('Value', 50);
  var small = boolean('Small Version', false);
  return React.createElement(ProgressBar, {
    value: value,
    small: small
  });
});
//# sourceMappingURL=ProgressBar.story.js.map