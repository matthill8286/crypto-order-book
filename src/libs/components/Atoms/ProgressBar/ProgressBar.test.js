import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { ProgressBar } from "./ProgressBar";
describe('ProgressBar', function () {
  it('renders basic', function () {
    var tree = renderWithTheme(React.createElement(ProgressBar, {
      value: 12
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=ProgressBar.test.js.map