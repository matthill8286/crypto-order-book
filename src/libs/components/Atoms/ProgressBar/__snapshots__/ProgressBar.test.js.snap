// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders basic 1`] = `
<progress
  className="ProgressBar__StyledProgressBar-iekkgs-0 celtLK"
  max={100}
  value={12}
/>
`;
